perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include "sim86_shared.h"
+#pragma comment (lib, "sim86_shared_debug.lib")
+#define ArrayCount(Array) (sizeof(Array) / sizeof((Array)[0]))
+
+s16 Registers[15] = {};
+
+char const* RegisterNames[][3] =
+{
+    {"", "", ""},
+    {"al", "ah", "ax"},
+    {"bl", "bh", "bx"},
+    {"cl", "ch", "cx"},
+    {"dl", "dh", "dx"},
+    {"sp", "sp", "sp"},
+    {"bp", "bp", "bp"},
+    {"si", "si", "si"},
+    {"di", "di", "di"},
+    {"es", "es", "es"},
+    {"cs", "cs", "cs"},
+    {"ss", "ss", "ss"},
+    {"ds", "ds", "ds"},
+    {"ip", "ip", "ip"},
+    {"flags", "flags", "flags"}
+};
+
+void SimulateInstruction(instruction Instruction)
+{
+    const char* Op = Sim86_MnemonicFromOperationType(Instruction.Op);
+    
+    if (Instruction.Op == Op_mov)
+    {
+        instruction_operand Dest = Instruction.Operands[0];
+        instruction_operand Source = Instruction.Operands[1];
+        s32 Value; // s32 to avoid issues on the bitshift right
+        switch (Source.Type)
+        {
+            case Operand_Register:
+            {
+                Value = Registers[Source.Register.Index];
+                if (Source.Register.Offset == 1) // Accessing high 8 bits
+                {
+                    Value >>= 8;
+                }
+                else if (Source.Register.Count == 1) // Accessing low 8 bits
+                {
+                    Value &= 0xFF;
+                }
+            } break;
+            case Operand_Immediate:
+            {
+                Value = Source.Immediate.Value;
+            } break;
+        }
+
+        Value = (s16)Value;
```

Comment:
It's a completely valid choice to use a C-style cast, but since some of your code has a C++ style, you might want to think about why you would prefer C-style casts or C++ style casts. Maybe you already thought about it and decided for C-style (which is fine)!

Reply:
I removed the need to cast here.
9e26986a2b2c67060d8c6cb56372b97fc66d96fc
I used C-style out of habit, I'll try to use C++ casts more to get used to them. Thanks!